controllers/nodefeaturediscovery_controls.go

```diff
@@ -41,25 +43,40 @@ const (
 	defaultServicePort int = 12000
 )
 
+// String returns the status of the resource as being Ready,
+// NotReady, or Unknown Resource Status
```

Comment:
I wouldn't reiterate the values in this comment. I suggest to be terse, e.g.:

```suggestion
// String implements the fmt.Stringer interface and returns describes
// ResourceStatus as a string.
```